### ISSUE TITLE
Roslynator changes the code logic when applying RSC1077 for negated patterns.

### ISSUE THREAD

**Issue by Yozer** [JosefPihrt/Roslynator#892]
**Product and Version Used**: 
Roslynator 4.1.0
VS 2022 17.1.2

**Steps to Reproduce**:
Apply suggested RSC1077:
![image](https://user-images.githubusercontent.com/9254709/161800421-a914055a-961b-46ac-a0a4-83616b4bd912.png)
on
```csharp
var list = Enumerable.Repeat(string.Empty, 10);
if (list.FirstOrDefault(t => t != "test") is not null)
    Console.WriteLine("ok");
else
    Console.WriteLine("bad");
```

**Actual Behavior**:
```csharp
var list = Enumerable.Repeat(string.Empty, 10);
if (!list.Any(t => t != "test"))
    Console.WriteLine("ok");
else
    Console.WriteLine("bad");
```

The condition is incorrectly negated.

**Expected Behavior**:
```csharp
var list = Enumerable.Repeat(string.Empty, 10);
if (list.Any(t => t != "test"))
    Console.WriteLine("ok");
else
    Console.WriteLine("bad");
```

Looks like Roslynator doesn't see negated pattern. It works fine when ```is not null``` is replaced with ```!= null```:
![image](https://user-images.githubusercontent.com/9254709/161801070-47234275-da43-4ef1-aa7b-741bdfebf31f.png)

Also, I noticed that RSC1077 is reported twice on this line:
![image](https://user-images.githubusercontent.com/9254709/161800920-353a20e8-b069-41ee-8765-d167f30e5426.png)

**Comment by Yozer** [comment]
Something similar but possibly unrelated:

Applying RSC1077 for on this also generate incorrect code:
```csharp
var list = Enumerable.Repeat((string)null, 10);
if (list.FirstOrDefault(t => t != "test") == null)
    Console.WriteLine("ok"); // this will print
else
    Console.WriteLine("bad");
```

actual (incorrect) output:
```csharp
var list = Enumerable.Repeat((string)null, 10);
if (!list.Any(t => t != "test"))
    Console.WriteLine("ok");
else
    Console.WriteLine("bad"); // this will print
```

**Comment by JosefPihrt** [comment]
Good catch! Thanks.
